docker/generate_build_matrix.py

```diff
@@ -194,6 +198,23 @@ def _main():
 
         pytorch_entries.append(entry)
 
+    nightly_entry = {
+        'AWS_OFI_NCCL_VERSION': '',
+        'BASE_IMAGE': 'nvidia/cuda:12.1.0-cudnn8-devel-ubuntu20.04',
+        'CUDA_VERSION': '12.1.0',
+        'IMAGE_NAME': 'torch-nightly-2-1-0-20230818-cu121',
+        'MOFED_VERSION': '5.5-1.0.3.2',
+        'PYTHON_VERSION': '3.10',
+        'PYTORCH_VERSION': '2.1.0',
+        'PYTORCH_NIGHTLY_URL': 'https://download.pytorch.org/whl/nightly/cu121',
+        'PYTORCH_NIGHTLY_VERSION': 'dev20230818+cu121',
+        'TAGS': ['mosaicml/pytorch_nightly:2.1.0_cu121-python3.10-ubuntu20.04'],
+        'TARGET': 'pytorch_stage',
+        'TORCHTEXT_VERSION': '0.16.0',
+        'TORCHVISION_VERSION': '0.16.0'
+    }
+    pytorch_entries.append(nightly_entry)
```

Comment:
Just flagging as we should remove this once official release comes out :)

Reply:
agreed :)